python/servo/try_parser.py

```diff
@@ -0,0 +1,171 @@
+# except according to those terms.
+
+import json
+import sys
+
+from dataclasses import dataclass
+from enum import Enum, Flag, auto
+
+
+class Layout(Flag):
+    none = 0
+    layout2013 = auto()
+    layout2020 = auto()
+
+    @staticmethod
+    def all():
+        return Layout.layout2013 | Layout.layout2020
+
+    def to_string(self):
+        if Layout.all() in self:
+            return "all"
+        elif Layout.layout2020 in self:
+            return "2020"
+        elif Layout.layout2013 in self:
+            return "2013"
+        else:
+            return "none"
+
+
+class Workflow(str, Enum):
+    LINUX = 'linux'
+    MACOS = 'macos'
+    WINDOWS = 'windows'
+    ANDROID = 'android'
+
+
+@dataclass
+class JobConfig(object):
+    name: str
+    workflow: Workflow = Workflow.LINUX
+    wpt_layout: Layout = Layout.none
+    profile: str = "release"
+    unit_tests: bool = False
+    wpt_tests_to_run: str = ""
+
+
+def handle_preset(s: str) -> JobConfig | None:
+    s = s.lower()
+
+    if s == "linux":
+        return JobConfig("Linux", Workflow.LINUX, unit_tests=True)
+    elif s in ["mac", "macos"]:
+        return JobConfig("MacOS", Workflow.MACOS, unit_tests=True)
+    elif s in ["win", "windows"]:
+        return JobConfig("Windows", Workflow.WINDOWS, unit_tests=True)
+    elif s in ["wpt", "linux-wpt"]:
+        return JobConfig("Linux WPT", Workflow.LINUX, unit_tests=True, wpt_layout=Layout.all())
+    elif s in ["wpt-2013", "linux-wpt-2013"]:
+        return JobConfig("Linux WPT legacy-layout", Workflow.LINUX, wpt_layout=Layout.layout2013)
+    elif s in ["wpt-2020", "linux-wpt-2020"]:
+        return JobConfig("Linux WPT layout-2020", Workflow.LINUX, wpt_layout=Layout.layout2020)
+    elif s in ["mac-wpt", "wpt-mac"]:
+        return JobConfig("MacOS WPT", Workflow.MACOS, wpt_layout=Layout.all())
+    elif s == "mac-wpt-2013":
+        return JobConfig("MacOS WPT legacy-layout", Workflow.MACOS, wpt_layout=Layout.layout2013)
+    elif s == "mac-wpt-2020":
+        return JobConfig("MacOS WPT layout-2020", Workflow.MACOS, wpt_layout=Layout.layout2020)
+    elif s == "android":
+        return JobConfig("Android", Workflow.ANDROID)
+    elif s == "webgpu":
+        return JobConfig("WebGPU CTS", Workflow.LINUX,
+                         wpt_layout=Layout.layout2020,  # reftests are mode for new layout
+                         wpt_tests_to_run="_webgpu",  # run only webgpu cts
+                         profile="production",  # WebGPU works to slow with debug assert
+                         unit_tests=False)  # production profile does not work with unit-tests
+    else:
+        return None
+
+
+class Encoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, (Config, JobConfig)):
+            return o.__dict__
+        if isinstance(o, Layout):
+            return o.to_string()
+        return json.JSONEncoder.default(self, o)
+
+
+class Config(object):
+    def __init__(self, s: str | None = None):
```

Comment:
```suggestion
    def __init__(self, s: Optional[str] = None):
```

Reply:
| is preferred: https://stackoverflow.com/questions/51710037/how-should-i-use-the-optional-type-hint, but I changed it anyway.